applications/crossbar/src/modules/cb_security.erl

```diff
@@ -0,0 +1,424 @@
+-spec delete(cb_context:context()) -> cb_context:context().
+delete(Context) ->
+    crossbar_doc:delete(Context, 'permanent').
+
+-spec delete(cb_context:context(), path_token()) -> cb_context:context().
+delete(Context, _) ->
+    io:format("~n after Doc ~p~n", [cb_context:doc(Context)]),
+    crossbar_doc:save(Context).
+
+%%--------------------------------------------------------------------
+%% @private
+%% @doc
+%% Attempt to load a summarized listing of all instances of this
+%% resource.
+%% @end
+%%--------------------------------------------------------------------
+-spec summary_available(cb_context:context()) -> cb_context:context().
+summary_available(Context) ->
+    cb_context:setters(Context, [{fun cb_context:set_resp_status/2, 'success'}
+                                ,{fun cb_context:set_resp_data/2, ?AVAILABLE_AUTH_MODULES}
+                                ]).
+
+%%--------------------------------------------------------------------
+%% @private
+%% @doc
+%% Load an instance from the database
+%% @end
+%%--------------------------------------------------------------------
+-spec read(cb_context:context()) -> cb_context:context().
+read(Context) ->
+    crossbar_doc:load(?ACCOUNT_AUTH_CONFIG_ID, Context, ?TYPE_CHECK_OPTION(<<"account_config">>)).
+
+-spec read(ne_binary(), cb_context:context()) -> cb_context:context().
+read(Id, Context) ->
+    C1 = crossbar_doc:load(?ACCOUNT_AUTH_CONFIG_ID, Context, ?TYPE_CHECK_OPTION(<<"account_config">>)),
+    case cb_context:resp_status(C1) of
+        'success' ->
+            case kz_json:get_value(module_config_path(Id), cb_context:doc(C1)) of
+                'undefined' -> not_found(Context, Id);
+                ModConfig ->crossbar_doc:handle_json_success(ModConfig, Context)
+            end;
+        _ -> not_found(Context, Id)
+    end.
+
+%%--------------------------------------------------------------------
+%% @private
+%% @doc
+%% Create a new instance with the data provided, if it is valid
+%% @end
+%%--------------------------------------------------------------------
+-spec create(cb_context:context()) -> cb_context:context().
+create(Context) ->
+    OnSuccess = fun(C) -> on_successful_validation('undefined', C) end,
+    cb_context:validate_request_data(<<"auth_configs">>, Context, OnSuccess).
+
+-spec create(ne_binary(), cb_context:context()) -> cb_context:context().
+create(Id, Context) ->
+    validate_patch(Id, Context).
+
+%%--------------------------------------------------------------------
+%% @private
+%% @doc
+%% Update an existing menu document with the data provided, if it is
+%% valid
+%% @end
+%%--------------------------------------------------------------------
+-spec update(ne_binary(), cb_context:context()) -> cb_context:context().
+update(?ACCOUNT_AUTH_CONFIG_ID=Id, Context) ->
+    OnSuccess = fun(C) -> on_successful_validation(Id, C) end,
+    cb_context:validate_request_data(<<"auth_configs">>, Context, OnSuccess);
```

Comment:
How about naming that schema `auth_config_account`?
1. it would have the word "account" in it
1. it would share a common prefix with its non-account counterpart